planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -2682,9 +2685,36 @@ TurnSignalInfo AvoidanceModule::calcTurnSignalInfo(const ShiftedPath & path) con
       turn_signal_info.turn_signal.command = TurnIndicatorsCommand::ENABLE_RIGHT;
     }
   } else {
-    turn_signal_info.turn_signal.command = TurnIndicatorsCommand::DISABLE;
+    const lanelet::ConstLanelets current_lanes = utils::getCurrentLanes(planner_data_);
+    const auto local_vehicle_footprint =
+      createVehicleFootprint(planner_data_->parameters.vehicle_info);
+    boost::geometry::model::ring<tier4_autoware_utils::Point2d> shifted_vehicle_footprint;
+    for (const auto & cl : current_lanes) {
+      const auto neighbor_lane = planner_data_->route_handler->getAllSharedLineStringLanelets(cl);
+      for (const auto & nl : neighbor_lane) {
+        // get left and right bounds of neighbor lane
+        const auto neighbor_left_bound = nl.leftBound2d().basicLineString();
+        const auto neighbor_right_bound = nl.rightBound2d().basicLineString();
+        for (size_t i = start_idx; i < end_idx; ++i) {
+          // transform vehicle footprint onto path points
+          shifted_vehicle_footprint = transformVector(
+            local_vehicle_footprint,
+            tier4_autoware_utils::pose2transform(path.path.points.at(i).point.pose));
+          if (segment_shift_length > 0.0) {
+            if (boost::geometry::intersects(neighbor_left_bound, shifted_vehicle_footprint)) {
+              turn_signal_info.turn_signal.command = TurnIndicatorsCommand::ENABLE_LEFT;
```

Comment:
You should consider the neighbor lanes direction if you would like to use its boundary line

Reply:
You can maybe consider implementing the second item on [my previous comment](https://github.com/autowarefoundation/autoware.universe/pull/4348#discussion_r1276237707) instead of using neighbor lane bounds.